core/rawdb/schema.go

```diff
@@ -66,6 +66,12 @@ var (
 	// fastTxLookupLimitKey tracks the transaction lookup limit during fast sync.
 	fastTxLookupLimitKey = []byte("FastTransactionLookupLimit")
 
+	// badBlockKey tracks the list of bad blocks seen by local
+	badBlockKey = []byte("BadBlock")
```

Comment:
This does clash with the `BloomBitsPrefix`, `B`. Is that OK? Just checking

Reply:
ping

Reply:
Sorry for the late reply. It's ok. The `badBlock` entry will share the same prefix `B`, but it's ok.
We have some addtional length check like 
`case bytes.HasPrefix(key, bloomBitsPrefix) && len(key) == (len(bloomBitsPrefix)+10+common.HashLength):`

So the badBlock won't be mixed into the bloomBits somehow. But I can change it to `InvalidBlock` if you want.

Reply:
done